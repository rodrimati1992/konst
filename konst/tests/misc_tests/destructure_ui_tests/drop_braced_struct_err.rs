struct Braced {
    bar: String,
    baz: String,
    qux: String,
}

impl Drop for Braced {
    fn drop(&mut self) {}
}


const fn to_bar(foo: Braced) -> (String, String, String) {
    konst::destructure!{Braced{bar, baz, qux} = foo}

    (bar, baz, qux)
}


fn main(){}