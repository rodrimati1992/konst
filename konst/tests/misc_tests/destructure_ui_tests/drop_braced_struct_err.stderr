error[E0308]: mismatched types
  --> tests/misc_tests/destructure_ui_tests/drop_braced_struct_err.rs:13:5
   |
13 |     konst::destructure!{Braced{bar, baz, qux} = foo}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `__DoesNotImplDrop<Braced>`, found `__ImplsDrop<Braced>`
   |
   = note: expected struct `__DoesNotImplDrop<Braced>`
              found struct `__ImplsDrop<Braced>`
   = note: this error originates in the macro `$crate::__destructure_struct` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)
